implementations/micrometer-registry-stackdriver/src/main/java/io/micrometer/stackdriver/StackdriverHistogramUtil.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2025 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.stackdriver;
+
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.distribution.*;
+
+final class StackdriverHistogramUtil {
+
+    private StackdriverHistogramUtil() {
+    }
+
+    // copied and modified from AbstractDistributionSummary/AbstractTimer
+    static Histogram stackdriverHistogram(Clock clock, DistributionStatisticConfig distributionStatisticConfig) {
+        if (distributionStatisticConfig.isPublishingPercentiles()) {
+            return new TimeWindowPercentileHistogram(clock, distributionStatisticConfig, true, false, true);
+        }
+        if (distributionStatisticConfig.isPublishingHistogram()) {
+            return new StackdriverFixedBoundaryHistogram(clock, distributionStatisticConfig);
+        }
+        return NoopHistogram.INSTANCE;
+    }
+
+    // Can't do this because java: cannot access org.HdrHistogram.DoubleRecorder
+    /*
+     * static class StackdriverClientSidePercentilesHistogram extends
+     * TimeWindowPercentileHistogram {
```

Comment:
This is what I wanted to do but the compiler won't let me. Adding a dependency to HdrHistogram to the stackdriver module didn't seem like the right thing to do. I wonder if we could make it a compile only dependency and if this would work as expected at runtime or not.

Reply:
Adding HdrHistogram as a compileOnly dependency allowed this to work as expected.